tensorflow/python/util/dispatch.py

```diff
@@ -43,6 +45,10 @@
 _GLOBAL_DISPATCHERS = []
 
 
+# Type variable for decorators
+F = TypeVar("F", bound=Callable[..., Any])
```

Comment:
should this be
```suggestion
F = TypeVar("F", bound=Callable[..., object])
```
perhaps there's no functional difference since its in a `TypeVar`, but it does eliminate an `Any`

Reply:
(btw, hi, I'm interested in helping out with adding type hints to tensorflow, so getting involved where I can. If this comment isnt helpful, feel free to ignore)

Reply:
Changing to `object` sounds reasonable to me.

It looks like this PR is still blocked on some merge conflicts. If would be helpful to unblock it by resolving the conflicts and making sure the integration tests still pass. This might need cloning it into a new PR.